build-tools/packages/build-tools/src/fluidBuild/fluidPackageCheck.ts

```diff
@@ -488,22 +490,57 @@ export class FluidPackageCheck {
     private static checkLintScripts(pkg: Package, fix: boolean) {
         let fixed = false;
         if (pkg.getScript("build")) {
-            const hasPrettier = pkg.getScript("prettier");
-            const lintChildren = hasPrettier ? ["prettier", "eslint"] : ["eslint"];
-            if (this.checkChildrenScripts(pkg, "lint", lintChildren, false, fix)) {
-                fixed = true;
+            // hasPrettier commented out to discard build warnings from "lint" & "lint:fix" scripts
+            // const hasPrettier = pkg.getScript("prettier");
+
+            // npm run lint
+            const lintScript = pkg.getScript("lint"); // acess lint-script
+            const lintPackages = lintScript?.match(/run\s*([^&]*)\s*/g); // regex to create a list of ["run foo", "run bar"]
+
+            if (lintPackages !== null) {
+                const lintChildren = lintPackages?.map((x) => x.trim().split(" ")[1]); // regex to create a list of ["foo", "bar"]
+
+                if (lintChildren !== undefined) {
+                    const hasEslint = lintChildren.some((elem) => /eslint/.test(elem)); // regex to check if "eslint" exists
+
+                    if (!hasEslint) {
+                        lintChildren?.push("eslint"); // push "eslint" to lintChildren as expected package
+                    }
+                }
+
+                if (this.checkChildrenScripts(pkg, "lint", lintChildren, false, fix)) {
+                    fixed = true;
+                }
             }
-            if (
-                this.checkChildrenScripts(
-                    pkg,
-                    "lint:fix",
-                    lintChildren.map((value) => `${value}:fix`),
-                    false,
-                    fix,
-                )
-            ) {
-                fixed = true;
+
+            // "npm run lint:fix"
+            const lintFixScript = pkg.getScript("lint:fix");
+            const lintFixPackages = lintFixScript?.match(/run\s*([^&]*)\s*/g);
+
+            if (lintFixPackages !== null) {
+                const lintFixChildren = lintFixPackages?.map((x) => x.trim().split(" ")[1]);
+
+                if (lintFixChildren !== undefined) {
+                    const hasEslint = lintFixChildren.some((elem) => /eslint/.test(elem));
```

Comment:
Should be regex here should be for `eslint:fix`, right?

Reply:
Ah, nice catch. Thank you for correcting!